### PR TITLE
add more data nodes

### PR DESCRIPTION
**Pull request by wybert**
This will solve #308

- [x] polish descriptions 
- [x] add test workflow

### CODE REVIEW

**Comment by koettert** [comment]
@koettert test that the data nodes especially the ESRI node also runs on Linux!

File: knime_extension/src/nodes/opendata.py [hunk 1]
```diff
@@ -686,3 +686,450 @@ def execute(self, exec_context: knext.ExecutionContext):
...
+        # TODO Create combined schema
+        return None
+
+    def execute(self, exec_context: knext.ExecutionContext):
+        import geopandas as gp
+        import pandas as pd
+        import requests
+
+        url = "https://opensky-network.org/api/states/all"
+        kws = {"url": url, "timeout": 120}
```
Comment (koettert) [inline knime_extension/src/nodes/opendata.py:793]: Expose timeout as advanced parameter

File: knime_extension/src/nodes/opendata.py [hunk 2]
```diff
@@ -686,3 +686,450 @@ def execute(self, exec_context: knext.ExecutionContext):
...
+        # TODO Create combined schema
+        return None
+
+    def execute(self, exec_context: knext.ExecutionContext):
+        import geopandas as gp
+        import pandas as pd
+        import requests
+
+        url = "https://ic-api.internetcomputer.org/api/v3/data-centers"
+        response = requests.get(url, timeout=120)
```
Comment (koettert) [inline knime_extension/src/nodes/opendata.py:863]: Expose timeout as advanced parameter

**Review by koettert — CHANGES_REQUESTED** [review]
I contacted our legal department to make sure that we are allowed to integrate the sources and under which terms. We need to make sure that we are allowed to include these sources into the AP.

**Comment by koettert** [comment]
@wybert when I use the geospatial_env.yml file of this PR I can no longer resolve my Python envrionment. It seems that it contains conflicting version.

**Comment by koettert** [comment]
Results from our legal department:
- Add a copyright notice similar to the OSM nodes to the GDelt node according to https://www.gdeltproject.org/about.html#termsofuse
- Remove the DataCenter node since it has no clear licensing
- For the Esri nodes we need to clearly show the copyright notice for each item in the KNIME table

**Comment by wybert** [comment]
> @wybert when I use the geospatial_env.yml file of this PR I can no longer resolve my Python envrionment. It seems that it contains conflicting version.

It got stuck in my Windows machine when creating a new environment.

**Comment by koettert** [comment]
image
Not sure if downgrading osmnx from version 1.7 (which we ship right now) to 1.2.2. What do you think?

**Comment by koettert** [comment]
To avoid dependency conflicts we will move the ESRI nodes out of the general Geospatial extension into a dedicated ESRI extension which can be installed separately. This extension would start to only provide the data nodes but could in the future be extended to provide nodes for more advanced and potentially commercial ArcGis features

**Comment by koettert** [comment]
@wybert please add a copyright notice similar to the OSM nodes to the GDelt node according to https://www.gdeltproject.org/about.html#termsofuse

File: knime_extension/geospatial_env.yml [hunk 3]
```diff
@@ -2,7 +2,9 @@ name: geospatial_env
 channels:
   - knime
   - conda-forge
+  - esri
```
Comment (koettert) [inline knime_extension/geospatial_env.yml:5]: @wybert please remove the dependency and once done assign to me

**Review by koettert — CHANGES_REQUESTED** [review]

**Pull request merged**